File: src/FamilyTree.Api/ClientApp/src/types/dateSuffix.ts
export interface DateSuffixDto {
  id: string;
  name: string;
  abbreviation: string;
  order: number;
  isDefault: boolean;
  worldId: string;
}

export interface CreateDateSuffixDto {
  name: string;
  abbreviation: string;
  order: number;
  isDefault: boolean;
  worldId: string;
}

export interface UpdateDateSuffixDto {
  name: string;
  abbreviation: string;
  order: number;
  isDefault: boolean;
}

File: src/FamilyTree.Domain/Entities/DateSuffix.cs
using FamilyTree.Domain.Common;

namespace FamilyTree.Domain.Entities;

public class DateSuffix : BaseEntity
{
    public required string Name { get; set; }
    public required string Abbreviation { get; set; }
    public int Order { get; set; }
    public bool IsDefault { get; set; }
    public Guid WorldId { get; set; }

    public World World { get; set; } = null!;
}
